text/0000-ppv.md

```diff
@@ -0,0 +1,1596 @@
+
+Note: While IEEE 754-2008 provides total ordering predicates for floating-point
+numbers, Rust does not implement `Eq` and `Ord` for the `f32` and `f64`
+primitive types. This RFC follows suit and does not propose to implement `Eq`
+and `Ord` for vectors of floating-point types. Any future RFC that might want to
+extend Rust with a total order for floats should extend the portable
+floating-point vector types with it as well. See [this internal
+thread](https://users.rust-lang.org/t/how-to-sort-a-vec-of-floats/2838/3) for
+more information.
+
+### Inherent Methods
+
+#### Construction and element access
+
+All portable signed integer, unsigned integer, and floating-point vector types
+implement the following methods:
+
+```rust
+impl {element_type}{lane_width}x{number_of_lanes} {
+/// Creates a new instance of the vector from `number_of_lanes` 
+/// values.
+pub const fn new(args...: element_type) -> Self;
+
+/// Returns the number of vector lanes.
+pub const fn lanes() -> usize;
+
+/// Constructs a new instance with each element initialized to
+/// `value`.
+pub const fn splat(value: element_type) -> Self;
+
+/// Extracts the value at `index`.
+///
+/// # Panics
+///
+/// If `index >= Self::lanes()`.
+pub fn extract(self, index: usize) -> element_type;
+
+/// Extracts the value at `index`.
+///
+/// If `index >= Self::lanes()` the behavior is undefined.
+pub unsafe fn extract_unchecked(self, index: usize) -> element_type;
+
+/// Returns a new vector where the value at `index` is replaced by `new_value`.
+///
+/// # Panics
+///
+/// If `index >= Self::lanes()`.
+#[must_use = error-message]
+pub fn replace(self, index: usize, new_value: $elem_ty) -> Self;
+
+/// Returns a new vector where the value at `index` is replaced by `new_value`.
+#[must_use = error-message]
+pub unsafe fn replace_unchecked(self, index: usize, 
+                                new_value: element_type) -> Self;
+}
+```
+
+#### Reads and Writes
+
+##### Contiguous reads and writes
+
+All portable vector types implement the following methods:
+
+```rust
+impl {element_type}{lane_width}x{number_of_lanes} {
+/// Writes the values of the vector to the `slice` without 
+/// reading or dropping the old value.
+///
+/// # Panics
+///
+/// If `slice.len() != Self::lanes()` or `&slice[0]` is not
+/// aligned to an `align_of::<Self>()` boundary.
+pub fn write_aligned(self, slice: &mut [element_type]);
+
+/// Writes the values of the vector to the `slice` without 
+/// reading or dropping the old value.
+///
+/// # Panics
+///
+/// If `slice.len() != Self::lanes()`.
+pub fn write_unaligned(self, slice: &mut [element_type]);
+
+/// Writes the values of the vector to the `slice` without 
+/// reading or dropping the old value.
+///
+/// # Precondition
+///
+/// If `slice.len() < Self::lanes()` or `&slice[0]` is not
+/// aligned to an `align_of::<Self>()` boundary, the behavior is
+/// undefined.
+pub unsafe fn write_aligned_unchecked(self, slice: &mut [element_type]);
```

Comment:
What's the added value of this method? We already have `ptr::write`. (Likewise for `read_aligned_unchecked` below.)

Plus, unlike `ptr::write` this API can't soundly be used for initializing memory: taking a `&mut` implies that the memory behind it must already be valid for `element_type` (technically this is all in-progress Unsafe Code Guidelines terriroty, but this principle seems pretty well established).

Reply:
> What's the added value of this method?

Now that I see the similarities with `pub unsafe fn write<T>(dst: *mut T, src: T)` where `T == vector type` I'd say that this adds very little value. AFAICT the only value these methods add is being able to switch between checked and unchecked variants while keeping the same API.